src/syntax/instrumentPpx.ml

```diff
@@ -87,14 +116,157 @@ let wrap_expr k e =
     else
       let marked = List.mem line c.CommentsPpx.marked_lines in
       let marker_file = !Location.input_name in
-      match marker marker_file ofs k marked with
+      match marker must_be_unique marker_file ofs k marked with
       | Some w -> Exp.sequence ~loc w e
       | None   -> e
 
+(* Given a pattern and a location, transforms the pattern into a list of
+   patterns by transforming or-patterns into cases (i.e., promoting them to the
+   top). Each resulting case pattern is paired with a list of locations to mark
+   should that case be reached. The location argument to this function is the
+   location of the closest enclosing or-pattern clause, or, if there is no such
+   clause, of the entire enclosing pattern. *)
+let translate_pattern =
```

Comment:
I think splitting it into 2 paragraphs might work, but I think I better leave it to you, since you have a fresher pair of eyes :)